src/AutomationTests/AutomationIntegrationTests.cs

```diff
@@ -94,54 +98,106 @@ public void Scan_Integration_WindowsFormsControlSampler()
             Scan_Integration_Core(WindowsFormsControlSamplerAppPath, WindowsFormsControlSamplerKnownErrorCount);
         }
 
+        [TestMethod]
+        [Timeout(30000)]
+        public void Scan_Integration_WindowsFormsMultiWindowSample()
+        {
+            Scan_Integration_Core(WindowsFormsMultiWindowSamplerAppPath, WindowsFormsMultiWindowSamplerAppAllErrorCount);
+        }
+
+        [TestMethod]
+        [Timeout(30000)]
+        public void Scan_Integration_WindowsFormsMultiWindowSample_SingleWindow()
+        {
+            Scan_Integration_Core(WindowsFormsMultiWindowSamplerAppPath, WindowsFormsMultiWindowSamplerSingleWindowAllErrorCount, false);
+        }
+
         [TestMethod]
         [Timeout(30000)]
         public void Scan_Integration_WpfControlSampler()
         {
             Scan_Integration_Core(WpfControlSamplerAppPath, WpfControlSamplerKnownErrorCount);
         }
 
-        private ScanResults Scan_Integration_Core(string testAppPath, int expectedErrorCount)
+        [TestMethod]
+        [Timeout(30000)]
+        public void SingleWindowScan_MultipleRootsEnabledThrows()
         {
-            LaunchTestApp(testAppPath);
+            LaunchTestApp(WindowsFormsMultiWindowSamplerAppPath);
+            var config = Config.Builder.ForProcessId(TestProcess.Id)
+                .WithOutputDirectory(OutputDir)
+                .WithOutputFileFormat(OutputFileFormat.A11yTest)
+                .WithMultipleScanRootsEnabled()
+                .Build();
+            var scanner = ScannerFactory.CreateScanner(config);
+
+            var action = new Action(() => scanner.Scan());
+            Assert.ThrowsException<InvalidOperationException>(action);
+        }
+
+        [TestMethod]
+        [Timeout(30000)]
+        public void SingleWindowScanWithID_MultipleRootsEnabledThrows()
+        {
+            LaunchTestApp(WindowsFormsMultiWindowSamplerAppPath);
             var config = Config.Builder.ForProcessId(TestProcess.Id)
                 .WithOutputDirectory(OutputDir)
                 .WithOutputFileFormat(OutputFileFormat.A11yTest)
+                .WithMultipleScanRootsEnabled()
                 .Build();
+            var scanner = ScannerFactory.CreateScanner(config);
+
+            var action = new Action(() => scanner.Scan("TestIDForThrow"));
+            Assert.ThrowsException<InvalidOperationException>(action);
+        }
+
+        private ScanResults Scan_Integration_Core(string testAppPath, int expectedErrorCount, bool enableMultipleScanRoots = true)
```

Comment:
Should this return `IReadOnlyCollection<ScanResults>` so that upstream tests can validate all roots? Maybe non-multiple root tests call a wrapper that sets enableMultipleScanRoots to false, asserts that only 1 ScanResult was returned, then returns the first element

Reply:
Currently, the result from this function is completely unused. I've updated the function to take an additional parameter that verifies how many outputs there are. Moving part of the verification outside the method and inside the actual tests would probably make the tests quite more complex and/or result in duplicate code. But if there is a need for this that with the updated behavior is not covered let me know.